relayer/src/services/relayer/optimism.ts

```diff
@@ -8,12 +14,21 @@ export class OptimismRelayerService extends BaseRelayerService {
   customNetworkContracts: DeepPartial<OEContractsLike> | undefined;
   messenger: CrossChainMessenger | undefined;
 
+  /** Used in the gas estimation process
+   *  https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/contracts/L1/OptimismPortal.sol#L417
+   *  https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/contracts/libraries/Constants.sol#L14-L21
+   *
+   *  This is the only way to catch reverts when relaying messages with incorrectly set minGasLimit. We also catch internal reverts from our contracts.
```

Comment:
I'm not 100% sure this is a correct description. By using the 0x1 address in the estimateGas call we basically force a revert in the contract, which then forces the estimate gas algorithm to also look at different execution paths adding more gas to the calculation. If the call is not being done from the 0x1 address, then upon hitting the check about the gasLimit the function just returns and we end up with a low gasLimit value.

I would go with something of the sort:

This is a "hack" in the optimism contracts. If the user provides a too low gasLimit the solidity call will fail. By faking the transaction as coming from the 0x1 address the optimism contracts are triggering a revert that causes the estimate algorithm to check all other possible execution paths and as such returns a higher/correct gasLimit value.

Reply:
>  If the user provides a too low gasLimit the solidity call will fail.

The call will actually not fail. The transaction will end successfully (and not even calling our internal message relay)